app/src/main/java/com/spascoding/taskycourse/core/data/remote/AuthInterceptor.kt

```diff
@@ -26,4 +52,62 @@ class AuthInterceptor @Inject constructor(
         return chain.proceed(request)
     }
 
+    private fun refreshToken(userInfo: UserInfo): String {
+        val refreshedToken = runBlocking(Dispatchers.IO) {
+            val request = RefreshTokenRequest(userInfo.refreshToken, userInfo.userId)
+            val response = RetrofitTokenClient.api.refreshAccessToken(request)
+            val body = response.body()
+            if (body != null) {
+                val resultUserInfo = userInfo.copy(
+                    accessToken = body.accessToken,
+                    accessTokenExpirationTimestamp = body.expirationTimestamp,
+                )
+                userInfoManager.saveUserInfo(resultUserInfo)
+                resultUserInfo.accessToken
+            } else {
+                userInfo.accessToken
+            }
+        }
+        Log.d("AuthInterceptor", "refreshedToken = $refreshedToken")
+        return refreshedToken
+    }
+
+    private object RetrofitTokenClient {
+        private const val BASE_URL = "https://tasky.pl-coding.com/"
+
+        private val client = OkHttpClient.Builder()
+            .addInterceptor(ApiKeyInterceptor())
+            .build()
+
+        val instance: Retrofit by lazy {
+            Retrofit.Builder()
+                .baseUrl(BASE_URL)
+                .client(client)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build()
+        }
+
+        val api: TaskyApiService by lazy {
+            instance.create(TaskyApiService::class.java)
+        }
+    }
+
+    private interface TaskyApiService {
+        @POST("/accessToken")
+        suspend fun refreshAccessToken(@Body request: RefreshTokenRequest): retrofit2.Response<RefreshTokenResponse>
+
+        /*
+        I don't know how to use this
+        Check authentication
+        Endpoint: /authenticate
+        Method: GET
+        Description: Used to check if a token is (still) valid
+        Response: 200 if valid, 401 if invalid
+         */
+//        @GET("/authenticate")
+//        suspend fun authenticate(
+//            @Header("Authorization") token: String
+//        ): retrofit2.Response<Void>
```

Comment:
You can use this to check during app launch if the saved token is still valid and keep the user logged in, but all in all this endpoint is optional since you can also check/refresh it with another endpoint